### PR TITLE
CLN: replace str.format() in pandas/io/formats

### PR DESCRIPTION
**Pull request by lithomas1**
- [x] ref #29547 and ref #30020 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Merging master should fix the CI issues.

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -1528,16 +1515,13 @@ def _maybe_wrap_formatter(formatter, na_rep: Optional[str]):
     elif callable(formatter):
         formatter_func = formatter
     else:
-        msg = (
-            "Expected a template string or callable, got {formatter} "
-            "instead".format(formatter=formatter)
-        )
+        msg = f"Expected a template string or callable, got {formatter} " "instead"
```
Comment (simonjayhawkins) [inline pandas/io/formats/style.py]:
```suggestion
        msg = f"Expected a template string or callable, got {formatter} instead"
```

File: pandas/io/formats/excel.py [hunk 2]
```diff
@@ -718,8 +716,8 @@ def write(
         if num_rows > self.max_rows or num_cols > self.max_cols:
             raise ValueError(
                 "This sheet is too large! Your sheet size is: "
-                + "{}, {} ".format(num_rows, num_cols)
-                + "Max sheet size is: {}, {}".format(self.max_rows, self.max_cols)
+                + f"{num_rows}, {num_cols} "
+                + f"Max sheet size is: {self.max_rows}, {self.max_cols}"
```
Comment (simonjayhawkins) [inline pandas/io/formats/excel.py]:
```suggestion
                f"Max sheet size is: {self.max_rows}, {self.max_cols}"
```

File: pandas/io/formats/excel.py [hunk 3]
```diff
@@ -718,8 +716,8 @@ def write(
         if num_rows > self.max_rows or num_cols > self.max_cols:
             raise ValueError(
                 "This sheet is too large! Your sheet size is: "
-                + "{}, {} ".format(num_rows, num_cols)
-                + "Max sheet size is: {}, {}".format(self.max_rows, self.max_cols)
+                + f"{num_rows}, {num_cols} "
```
Comment (simonjayhawkins) [inline pandas/io/formats/excel.py]:
```suggestion
                f"{num_rows}, {num_cols} "
```

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @lithomas1 for the PR. couple of suggestions otherwise lgtm.

**Review by WillAyd — APPROVED** [review]
lgtm outside of suggestions from @simonjayhawkins

**Pull request closed without merging**